platform/core/src/capabilities/CreationCapability.js

```diff
@@ -0,0 +1,79 @@
+ * Administration. All rights reserved.
+ *
+ * Open MCT Web is licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * Open MCT Web includes source code licensed under additional open source
+ * licenses. See the Open Source Licenses file (LICENSES.md) included with
+ * this source code distribution or the Licensing information page available
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+/*global define,Promise*/
+
+define(
+    ['../objects/DomainObjectImpl', 'uuid'],
+    function (DomainObjectImpl, uuid) {
+        'use strict';
+
+        /**
+         * Implements the `creation` capability. This allows new domain
+         * objects to be instantiated.
+         *
+         * @constructor
+         * @memberof platform/core
+         * @param $injector Angular's `$injector`
+         */
+        function CreationCapability($injector) {
+            this.$injector = $injector;
+        }
+
+        /**
+         * Alias of `capabilityService.getCapabilities`; handles lazy loading
+         * of `capabilityService`, since it cannot be declared as a
+         * dependency directly without creating a cycle.
+         * @private
+         */
+        CreationCapability.prototype.getCapabilities = function (model) {
+            if (!this.capabilityService) {
+                this.capabilityService =
+                    this.$injector.get('capabilityService');
+            }
+            return this.capabilityService.getCapabilities(model);
+        };
+
+        /**
+         * Instantiate a new domain object with the provided model.
+         *
+         * This domain object will have been simply instantiated; it will not
+         * have been persisted, nor will it have been added to the
+         * composition of the object which exposed this capability.
+         *
+         * @returns {DomainObject} the new domain object
+         */
+        CreationCapability.prototype.create = function (model) {
+            var id = uuid(),
+                capabilities = this.getCapabilities(model);
+            return new DomainObjectImpl(id, model, capabilities);
+        };
+
+        /**
+         * Alias of `create`.
+         * @see {platform/core.CreationCapability#create}
+         */
+        CreationCapability.prototype.invoke =
+            CreationCapability.prototype.create;
+
+        return CreationCapability;
+    }
+);
```

Comment:
Hrm, ok I understand the rationale now, thanks. 
Here's a specific case where I think it gets tricky though -
* In new edit mode, the user can drag a plot onto another plot which has the effect of creating a new 'virtual' panel in memory. 
* The view switches to edit mode immediately, and the location of the new virtual panel is not determined until the user hits save, at which point the create dialog is launched allowing them to select a location in the object tree to save their new panel. 
* The two original plots might have been persisted in a remote elastic search instance, but the user might want to persist their new combined panel into My Items. 
* The virtual panel could be created initially using the persistence space of the base plot, but this would then need to be updated prior to persistence.